### PR TITLE
Balder/reduce includes and visibility

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR
No semantic changes, just less includes and less visibility

### CODE REVIEW

File: staging_vespalib/src/vespa/vespalib/util/shutdownguard.cpp [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
 
 #include <vespa/log/log.h>
 LOG_SETUP(".vespalib.shutdownguard");
 
 namespace vespalib {
 
+namespace {
+enum { STACK_SIZE = (1u << 16) };
+
+static uint64_t getTimeInMillis() {
```
Comment (vekterli) [inline staging_vespalib/src/vespa/vespalib/util/shutdownguard.cpp]: Intentionally inherited `static`-ness from when it was inside class `ShutdownGuard`?

**Review by vekterli — APPROVED** [review]
LGTM. Have a couple of questions and minor suggestions for follow-up

File: fastos/src/vespa/fastos/file.cpp [hunk 2]
```diff
@@ -495,3 +496,18 @@ bool FastOS_FileInterface::Rename (const char *newFileName)
...
+FastOS_DirectoryScanInterface::FastOS_DirectoryScanInterface(const char *path)
+    : _searchPath(strdup(path))
+{
+}
+
+/**
+ * Destructor.
+ *
+ * Frees operating system resources related to the directory scan.
+ */
```
Comment (baldersheim) [inline fastos/src/vespa/fastos/file.cpp]: Fixed

File: fastos/src/vespa/fastos/unix_dynamiclibrary.cpp [hunk 3]
```diff
@@ -56,7 +56,7 @@ FastOS_UNIX_DynamicLibrary::Close()
     bool retcode = true;
 
     if (IsOpen()) {
-        retcode = (dlclose(_handle) == TRUE);
+        retcode = (dlclose(_handle) == 0);
```
Comment (baldersheim) [inline fastos/src/vespa/fastos/unix_dynamiclibrary.cpp]: I also shook my eyes on this one. Either one one checks the return code or TRUE is 0.....

File: fastos/src/vespa/fastos/unix_process.cpp [hunk 4]
```diff
@@ -1408,6 +1413,29 @@ bool FastOS_UNIX_ProcessStarter::CreateSocketPairs ()
     return rc;
 }
 
+FastOS_UNIX_ProcessStarter::FastOS_UNIX_ProcessStarter (FastOS_ApplicationInterface *app)
+    : _app(app),
+      _processList(NULL),
```
Comment (baldersheim) [inline fastos/src/vespa/fastos/unix_process.cpp]: Fixed

File: fastos/src/vespa/fastos/unix_process.cpp [hunk 5]
```diff
@@ -1979,3 +2007,41 @@ bool FastOS_UNIX_ProcessStarter::Detach(FastOS_UNIX_Process *process)
...
+    _canRead = false;
+    _canWrite = false;
+    _pollIdx = -1;
+    if (_fd != -1) {
+        close(_fd);
+        _fd = -1;
+    }
+    if (_readBuffer.get() != NULL)
+        _readBuffer->Close();
+    if (_writeBuffer.get() != NULL)
```
Comment (baldersheim) [inline fastos/src/vespa/fastos/unix_process.cpp]: Fixed.

File: staging_vespalib/src/vespa/vespalib/util/shutdownguard.cpp (hunk 1 above)
Comment (baldersheim) [inline staging_vespalib/src/vespa/vespalib/util/shutdownguard.cpp]: Copy&paste without intelligent filters....

**Pull request merged**